content/document.js

```diff
@@ -213,9 +213,29 @@ const findByURL = memoize((url, fields = null) => {
   return document ? { contentRoot: CONTENT_ROOT, folder, document } : null;
 });
 
-function findAll() {
+function findAll(
+  { files, foldersearch } = { files: new Set(), foldersearch: null }
```

Comment:
nit: `foldersearch` casing is off. Maybe I'll also prefer something like `folderPattern` (though I guess that suggests that it already supports regex).